src/interactive-window/editor-integration/codeGenerator.ts

```diff
@@ -298,24 +290,28 @@ export class CodeGenerator implements IInteractiveWindowCodeGenerator {
         source: string[],
         trueStartLine: number,
         firstNonBlankLineIndex: number,
+        hasCellMarker: boolean,
         usingJupyterDebugProtocol?: boolean
     ): { runtimeLine: number; debuggerStartLine: number } {
         const useNewDebugger =
-            usingJupyterDebugProtocol || this.configService.getSettings(undefined).useJupyterDebugger === true;
+            usingJupyterDebugProtocol || this.configService.getSettings(undefined).forceIPyKernelDebugger === true;
         if (debug && this.configService.getSettings(this.notebook.uri).stopOnFirstLineWhileDebugging) {
             if (useNewDebugger) {
-                return { runtimeLine: 1, debuggerStartLine: firstNonBlankLineIndex + 1 };
+                // Inject the breakpoint line
+                source.splice(0, 0, 'breakpoint()\n');
+                return { runtimeLine: 1, debuggerStartLine: trueStartLine + 1 };
```

Comment:
Since we are injecting the bp here I thought that we needed to start one line past.

Reply:
> Since we are injecting the bp here I thought that we needed to start one line past.

Yes thats' what happens, we start in the line after the breakpoint, hence the `trueStartLine + 1`

Reply:
Got'cha. I think that I'm confusing runtimeLine and debuggerStartLine's usage some. Will look it over again.